scripts/_common.sh

```diff
@@ -4,34 +4,31 @@
 # COMMON VARIABLES
 #=================================================
 
-YNH_PHP_VERSION="7.4"
-
-# dependencies used by the app
-pkg_dependencies="ntp ca-certificates unzip curl cron \
-locate tar telnet wget logrotate fail2ban dos2unix ntpdate htop iotop vim iftop smbclient \
-git \
-python3 python3-pip \
-software-properties-common \
-libexpat1 ssl-cert \
-apt-transport-https \
-xvfb cutycapt xauth \
-duplicity \
-at \
-php$YNH_PHP_VERSION-curl php$YNH_PHP_VERSION-gd php$YNH_PHP_VERSION-imap php$YNH_PHP_VERSION-xml php$YNH_PHP_VERSION-soap php$YNH_PHP_VERSION-xmlrpc php$YNH_PHP_VERSION-dev php$YNH_PHP_VERSION-zip php$YNH_PHP_VERSION-ssh2 php$YNH_PHP_VERSION-mbstring php$YNH_PHP_VERSION-ldap \
-ffmpeg"
-
-# ntp usb-modeswitch python-serial
-extra_pkg_dependencies="libsox-fmt-mp3 sox libttspico-utils \
-espeak \
-mbrola \
-brltty \
-net-tools \
-nmap"
-
 #=================================================
 # PERSONAL HELPERS
 #=================================================
 
+_jeedom_set_permission() {
+    # For nginx user to be able to communicate with home automation devices
+    if getent group dialout >/dev/null; then
+        usermod -a -G dialout "$app"
+    fi
+    if getent group gpio >/dev/null; then
+        usermod -a -G gpio "$app"
+    fi
+    if getent group tty >/dev/null; then
+        usermod -a -G tty "$app"
+    fi
+
+    # Sudo permissions to the user if agreed to
+    if [ "$sudo" ]; then
+        usermod -a -G "sudo" "$app"
+        if ! grep -q "www-data ALL=(ALL) NOPASSWD: ALL" /etc/sudoers; then
+            echo "www-data ALL=(ALL) NOPASSWD: ALL" | (EDITOR="tee -a" visudo)
+        fi
```

Comment:
Wtf, is this really adding www-data to sudoers like it's no biggy ? x_x This is a security disaster, it can't even be undone in a safe way ... Why would we need to have the entire www-data group to be sudoer x_x

Reply:
OMG, didn't even see that. Jeedom devs doing sudo like nothing's wrong… ffs. Also maybe the user only should be in sudoers ?

Reply:
Yes, definitely only the user should be in sudo in "worst case scenario" ... But i'm also even puzzled why it needs to run sudo in the first place ...